### ISSUE TITLE
Cannot enter custom command pallette command

### ISSUE THREAD

**Issue by AdamLeMmon01** [sublimehq/sublime_merge#1489]
**Version info**

 - OS: macOS 12.3.1
 - Build: 2071

**Description**

When I have this Default.sublime-commands file, it does not show in the command pallette. If I use the git command instead of gh, it recognizes it just fine. Am I not able to enter arbitrary custom commands here? It seems I'm only able to add git as the command with various args

**Steps to reproduce**

1. Open prefs -> browse packages -> navigate to user folder
2. create Default.sublime-commands file
3. enter the following text
[
  {
    "caption": "Open Pull Request",
    "command": "gh",
    "args": {"argv": ["pr", "list"]}
  }
]
4. save
5. open command pallet and type "Open P..." note that the custom command does not appear

**Expected behavior**

I should be able to run arbitrary commandline commands here or it's not very flexible.

**Comment by Ultra-Instinct-05** [comment]
> If I use the git command instead of gh, it recognizes it just fine. Am I not able to enter arbitrary custom commands here? It seems I'm only able to add git as the command with various args

Yes, because SM doesn't have the concept of plugin defined commands. Only ST has that. You can only use commands defined by SM only and can't construct arbitrary ones

**Comment by dpjohnst** [comment]
Hi @AdamLeMmon01,

Thanks for reaching out!

As mentioned by @Ultra-Instinct-05, there is only the concept of predefined commands in Sublime Merge. We do offer the `git` command, which allows execution of arbitrary `git` commands. This does not extend to arbitrary programs.

In the meantime, Git does allow you to define aliases, in which you can shell out to execute arbitrary programs (such as `gh`).
For example, you could configure the following alias:
`git config --global alias.pr-list "!f() { gh pr list; }; f"`

You can then treat the above as if it's a `git` command - executing this in Sublime Merge would require you to add the `git` command with the argument `pr-list` to the `Default.sublime-commands` file.

Hope this helps!
\- Dylan

**Comment by AdamLeMmon01** [comment]
This works well enough for my needs. It's kind of roundabout and could use some better documentation

**Comment by srbs** [comment]
@AdamLeMmon01, instead of creating an alias per command you want, a more flexible approach would be to use this:

```git config --global alias.gh '!f() { gh "$@"; }; f'```

Then each command/menu item can use this approach:
```
{
    "caption": "Open Pull Request",
    "command": "git",
    "args": {"argv": ["gh", "pr", "list"]}
}
```
